tests/unit/test_synthDrivers/test_espeak.py

```diff
@@ -0,0 +1,139 @@
+			msg="Language without locale used when available"
+		)
+		self.assertEqual(
+			LangChangeCommand("ta-ta"),
+			SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand("ta-gb")),
+			msg="Language with any locale used when available"
+		)
+		with self.assertLogs(logHandler.log, level=logging.DEBUG) as logContext:
+			self.assertEqual(
+				LangChangeCommand(None),
+				SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand("fake")),
+				msg="No matching available language returns None"
+			)
+		self.assertIn(
+			"Unable to find an eSpeak language for 'fake'",
+			logContext.output[0]
+		)
+
+
+class TestSynthDriver_Integration(unittest.TestCase):
+	"""Perform integration testing for the eSpeak synth driver."""
+
+	def setUp(self) -> None:
+		# Prevent NVWave from causing an exception
+		# when running unit tests in AppVeyor
+		self._nvwaveOpenOld = nvwave.WavePlayer.open
+		nvwave.WavePlayer.open = lambda self: None
+		self._driver = SynthDriver()
+	
+	def tearDown(self) -> None:
+		self._driver.terminate()
+		nvwave.WavePlayer.open = self._nvwaveOpenOld
+
+	def test_defaultMappingAvailableLanguage(self):
+		"""Confirms language codes remapped by default are supported by eSpeak and are comprehensive."""
+		eSpeakAvailableLangs = self._driver.availableLanguages
+		mappedDefaultLanguages = set(self._driver._defaultLangToLocale.values())
+		unexpectedUnsupportedDefaultLanguages = mappedDefaultLanguages.difference(eSpeakAvailableLangs)
+		self.assertEqual(
+			set(),
+			unexpectedUnsupportedDefaultLanguages,
+			msg=(
+				"Languages mapped by default are no longer supported by eSpeak: "
+				f"{unexpectedUnsupportedDefaultLanguages}"
+			)
+		)
+
+		expectedUnsupportedMappedLanguages = set(self._driver._defaultLangToLocale.keys())
+		unexpectedSupportedMappedLanguages = expectedUnsupportedMappedLanguages.intersection(eSpeakAvailableLangs)
+		self.assertEqual(
+			set(),
+			unexpectedSupportedMappedLanguages,
+			msg=(
+				"Languages mapped to eSpeak defaults are now supported by eSpeak: "
+				f"{unexpectedSupportedMappedLanguages}"
+			)
+		)
+
+		supportedLanguagesWithLocaleStripped = set(lang.split("-")[0] for lang in eSpeakAvailableLangs)
+		unsupportedLanguagesWithoutLocale = supportedLanguagesWithLocaleStripped.difference(eSpeakAvailableLangs)
+		self.assertEqual(
+			expectedUnsupportedMappedLanguages,
+			unsupportedLanguagesWithoutLocale,
+			msg=(
+				"eSpeak has a language with locale supported"
+				"but the language without locale is unsupported."
+				"Update _defaultLangToLocale to include a new mapping."
+			)
+		)
+
+	def test_availableLanguagesWithoutLocale(self):
+		"""
+		Confirms that eSpeak can manually be switched to all of its supported languages, with the locale removed.
+		This doesn't test automatic language switching.
+		"""
+		availableLanguagesWithoutLocale = set(lang.split("-")[0] for lang in self._driver.availableLanguages)
```

Comment:
This `lang.split("-")[0]` is repeated many times. Can you move it to a helper function.

```python
def _getLangCodeOnly(langWithOptionalLocale: str) -> str:
	""" Get the lang code eg "en" for "en-au" or "chr" for "chr-US-Qaaa-x-west"
	@param langWithOptionalLocale: may already be language only, or include locale specifier. EG "en" or "en-au"
	@return The language only part, before the first dash. 
	"""
	return langWithOptionalLocale.split("-")[0]

Reply:
I've done this in languageHandler